core/src/main/scala/org/apache/spark/sql/delta/commands/CreateDeltaTableCommand.scala

```diff
@@ -295,7 +295,7 @@ case class CreateDeltaTableCommand(
     // Otherwise we'll just go with the metadata already present in the log.
     // The schema compatibility checks will be made in `WriteIntoDelta` for CreateTable
     // with a query
-    if (txn.readVersion > -1) {
+    if (txn.deltaLog.tableExists) {
```

Comment:
is it guaranteed that `txn.deltaLog.snapshot` is the same as `txn.snapshot`?
i think it's safer to keep using `txn.readVersion > -1`